File: Arbitration/MPArbitration/ClientApp/src/app/model/case-arbitrator.ts
import { LogLevel } from "@azure/msal-browser";
import { loggerCallback } from "../app.module";
import { Arbitrator, IArbitrator, IArbStats } from "./arbitrator";
import { IModifier } from "./imodifier";

export enum ArbitratorDisqualification {
    Arbitrator,
    None,
    Payor,
    Provider,
    TDI
}

export class CaseArbitrator implements IArbitrator, IModifier {
    public id = 0;
    public allStats: IArbStats[] = [];
    public arbitrationCaseId: number | undefined;
    public arbitratorId: number | undefined;
    public arbitrator: Arbitrator | undefined;
    public assignedOn: Date|undefined;
    public disqualifiedBy = ArbitratorDisqualification.None;
    private eliminateForServices = '';
    public email = ''; // viewmodel property
    public fee = 0;
    public isActive = true;
    public isLastResort = false; // viewmodel property
    public isDismissed = false;
    public name = ''; // viewmodel property
    public notes = ''; // viewmodel property
    public phone = ''; // viewmodel property
    public statistics = ''; // viewmodel property
    public updatedBy = '';
    public updatedOn: Date|undefined;

    constructor(obj?:any) {
        if(!obj)
            return;
        Object.assign(this, JSON.parse(JSON.stringify(obj)));
        if(this.updatedOn)
            this.updatedOn = new Date(obj.updatedOn); 
        if(this.arbitrator)
            this.arbitrator = new Arbitrator(this.arbitrator);
            
        if(this.statistics) {
            // parse stats
            try {
                this.allStats = JSON.parse(this.statistics);
            } catch {
                loggerCallback(LogLevel.Warning,'Unable to parse Arbitrator Statistics');
            }
        }    
    }
    
    get cases():number {
        let sum=0;
        this.allStats.map(d => sum+=d.cases);
        return sum;
    }
    get losses():number {
        let sum=0;
        this.allStats.map(d => sum+=d.lost);
        return sum;
    }
    get wins():number {
        let sum=0;
        this.allStats.map(d => sum+=d.won);
        return sum;
    }
    get unfavorableServices():string[] {
        return this.eliminateForServices.split(';');
    }
}

File: Arbitration/MPArbitration/Model/CaseArbitrator.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    public enum ArbitratorDisqualification
    {
        Arbitrator,  // arbitrator disqualifies themselves (sickness, etc - no idea if this happens but seems possible)
        None, // not disqualified
        Payor,  // payor disqualified the arbitrator
        Provider,  // provider disqualified the arbitrator
        TDI  // TDI disqualified the arbitrator for some reason (no idea if this happens but seems possible)
    }

    public class CaseArbitrator
    {
        [JsonPropertyName("id")]
        public int Id { get; set; } = 0;

        [JsonPropertyName("assignedOn")]
        public DateTime? AssignedOn { get; set; } = null;

        [JsonPropertyName("disqualifiedBy")]
        [Column(TypeName = "nvarchar(60)")]
        public ArbitratorDisqualification DisqualifiedBy { get; set; }

        [NotMapped]
        [JsonPropertyName("eliminateForServices")]
        public string EliminateForServices { get; set; } = "";

        [NotMapped]
        [JsonPropertyName("email")]
        [StringLength(60)]
        public string Email { get; set; } = "";

        [JsonPropertyName("fee")]
        public double Fee { get; set; } = 0;

        [JsonPropertyName("isActive")]
        public bool IsActive { get; set; }

        [NotMapped]
        [JsonPropertyName("isLastResort")]
        public bool IsLastResort { get; set; }

        [JsonPropertyName("isDismissed")]
        public bool IsDismissed { get; set; }

        [NotMapped]
        [JsonPropertyName("name")]
        public string Name { get; set; } = "";

        [NotMapped]
        [JsonPropertyName("notes")]
        public string Notes { get; set; } = "";

        [NotMapped]
        [JsonPropertyName("phone")]
        public string Phone { get; set; } = "";

        [NotMapped]
        [JsonPropertyName("statistics")]
        public string Statistics { get; set; } = "";

        [JsonPropertyName("updatedBy")]
        [StringLength(60)]
        public string UpdatedBy { get; set; } = "";

        [JsonPropertyName("updatedOn")]
        public DateTime? UpdatedOn { get; set; } = null;

        [JsonPropertyName("arbitratorId")]
        public int? ArbitratorId { get; set; } = null;

        [JsonPropertyName("arbitrationCaseId")]
        public int? ArbitrationCaseId { get; set; } = null;

        [NotMapped]
        [JsonPropertyName("arbitrator")]
        public virtual Arbitrator? Arbitrator { get; set; } = null;
    }
}
